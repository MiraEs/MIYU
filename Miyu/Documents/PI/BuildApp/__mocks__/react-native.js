/* eslint-disable react/no-multi-comp */
import {
	createElement,
	Component,
} from 'react';
import PropTypes from 'prop-types';
const ReactNative = jest.genMockFromModule('react-native');

ReactNative.StyleSheet = {
	create: function create(styles) {
		return styles;
	},
	flatten: function flatten(...styles) {
		return Object.assign({}, ...styles);
	},
};

ReactNative.PixelRatio = {
	get() {
		return 1;
	},
};

ReactNative.Alert = {
	alert: jest.fn(),
};

ReactNative.StatusBar = {
	setBarStyle: jest.fn(),
	setHidden: jest.fn(),
};

ReactNative.Linking = {
	openURL: jest.fn(),
};

ReactNative.NativeModules = {
	PhotoUploader: {},
	SafariViewManager: {},
	UIManager: {
		measure: jest.fn(),
	},
	MarketingCloud: {
		trackAction: jest.fn(),
		trackState: jest.fn(),
	},
	LocalyticsManager: {
		tagScreen: jest.fn(),
		tagEvent: jest.fn(),
		tagCustomerLoggedOut: jest.fn(),
	},
	Instabug: {
		log: jest.fn(),
	},
	BNDevice: {},
	RNApplePayManager: {
		canMakePayments: jest.fn(() => Promise.resolve(true)),
		canMakePaymentsUsingNetworks: jest.fn(() => Promise.resolve(true)),
		paymentRequest: jest.fn(),
		paymentRequestDidFinish: jest.fn(),
		updateShippingMethods: jest.fn(),
		selectShippingMethod: jest.fn(),
		authorizedPayment: jest.fn(),
	},
};

ReactNative.requireNativeComponent = jest.fn();

ReactNative.NetInfo = {
	isConnected: {
		fetch: jest.fn(() => ({ done: jest.fn() })),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
	},
};

ReactNative.AppState = {
	addEventListener: jest.fn(),
};

ReactNative.CameraRoll = {
	getPhotos: jest.fn(() => Promise.resolve([])),
};

ReactNative.PushNotificationIOS = {
	setApplicationIconBadgeNumber: jest.fn(),
};

ReactNative.Keyboard = {
	dismiss: jest.fn(),
};

ReactNative.findNodeHandle = jest.fn();

ReactNative.ViewPropTypes = {
	style: jest.fn(),
};

ReactNative.NativeEventEmitter = class NativeEventEmitter {
	addListener = jest.fn(() => ({ remove: jest.fn() }));
};

class FlatList extends Component {
	render() {
		return createElement('FlatList', this.props);
	}
}

class SectionList extends Component {
	render() {
		return createElement('SectionList', this.props);
	}
}

class View extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('View', this.props, this.props.children);
	}
}

class ScrollView extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('ScrollView', this.props, this.props.children);
	}
}

class Switch extends Component {
	render() {
		return createElement('Switch');
	}
}

class Text extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Text', this.props, this.props.children);
	}
}

class TextInput extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TextInput', this.props, this.props.children);
	}
}

class TouchableOpacity extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableOpacity', this.props, this.props.children);
	}
}

class TouchableWithoutFeedback extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableWithoutFeedback', this.props, this.props.children);
	}
}

class ToolbarAndroid extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ToolbarAndroid', this.props, this.props.children);
	}
}

class Image extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Image', this.props, this.props.children);
	}
}

class Icon extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('Icon', this.props, this.props.children);
	}
}

class ActivityIndicator extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ActivityIndicator', this.props, this.props.children);
	}
}

ReactNative.Platform = {
	OS: 'ios',
	select: jest.fn((options) => options.ios),
};

ReactNative.InteractionManager = {
	runAfterInteractions: jest.fn(),
};

ReactNative.ActionSheetIOS = {
	showActionSheetWithOptions: jest.fn(),
};

class ListView extends Component {
	DataSource() {
		return {
			cloneWithRows: jest.fn(),
			cloneWithRowsAndSections: jest.fn(),
		};
	}
}

class AppRegistry {
	registerComponent() {}
}

class TouchableHighlight extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableHighlight', this.props, this.props.children);
	}
}
class BackHandler extends Component {
	static addEventListener() {}
	static exitApp() {}
	static removeEventListener() {}
}

class TouchableNativeFeedback extends Component {
	static Ripple() {
	}
}

class Navigator extends Component {
	render() {
	}
}

class RefreshControl extends Component {
	render() {
		return createElement('RefreshControl', this.props);
	}
}

class LayoutAnimation extends Component {
	static configureNext() {
	};

	static Presets() {
		return {
			easeInEaseOut: jest.fn(),
		};
	};
}

class Dimensions {
	static get() {
		return 0;
	}
}

class Animated extends Component {

	static Value() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static ValueXY() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static event() {}
}

class AnimatedView extends Component {
	render() {
		return createElement('AnimatedView');
	}
}

class AnimatedImage extends Component {

}

class AnimatedText extends Component {

}

Animated.View = AnimatedView;

Animated.Image = AnimatedImage;

Animated.Text = AnimatedText;

class PanResponder {
	static create() {
		return {panHandlers: {}};
	}
}

global.__fbBatchedBridgeConfig = {
	remoteModuleConfig: [],
	localModulesConfig: [],
};

ReactNative.View = View;
ReactNative.ScrollView = ScrollView;
ReactNative.FlatList = FlatList;
ReactNative.ListView = ListView;
ReactNative.Text = Text;
ReactNative.TextInput = TextInput;
ReactNative.TouchableOpacity = TouchableOpacity;
ReactNative.TouchableHighlight = TouchableHighlight;
ReactNative.TouchableWithoutFeedback = TouchableWithoutFeedback;
ReactNative.ToolbarAndroid = ToolbarAndroid;
ReactNative.Image = Image;
ReactNative.AppRegistry = AppRegistry;
ReactNative.Icon = Icon;
ReactNative.ActivityIndicator = ActivityIndicator;
ReactNative.BackHandler = BackHandler;
ReactNative.TouchableNativeFeedback = TouchableNativeFeedback;
ReactNative.Navigator = Navigator;
ReactNative.RefreshControl = RefreshControl;
ReactNative.LayoutAnimation = LayoutAnimation;
ReactNative.Dimensions = Dimensions;
ReactNative.Animated = Animated;
ReactNative.PanResponder = PanResponder;
ReactNative.SectionList = SectionList;
ReactNative.Switch = Switch;

module.exports = ReactNative;
