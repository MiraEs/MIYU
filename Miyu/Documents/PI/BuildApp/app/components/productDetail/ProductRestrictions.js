
import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	StyleSheet,
	Text,
	View,
} from 'react-native';
import { connect } from 'react-redux';
import InlineAlert from '../InlineAlert';
import pluralize from 'pluralize';
import styles from '../../lib/styles';
import Button from '../button';
import TrackingActions from '../../lib/analytics/TrackingActions';
import productConfigurationHelpers from '../../lib/ProductConfigurationHelpers';

const componentStyles = StyleSheet.create({
	inlineAlert: {
		margin: styles.measurements.gridSpace1,
	},
	bodyText: {
		textAlign: 'center',
		color: styles.colors.error,
	},
	button: {
		marginTop: styles.measurements.gridSpace1,
	},
});

export class ProductRestrictions extends Component {

	navigateToSearch = () => {
		const keyword = this.props.application;
		this.props.navigator.push('productDrops', {
			searchCriteria: {
				keyword,
				page: 1,
				pageSize: 50,
			},
			tracking: {
				name: 'build:app:productrestrictions:searchresults',
				data: {
					keyword,
				},
			},
		});
	};

	renderButton = (restriction) => {
		if (restriction.policyDescription.toLowerCase().indexOf('ab1953') !== -1) {
			return (
				<Button
					text="Shop Similar Products"
					onPress={this.navigateToSearch}
					style={componentStyles.button}
					trackAction={TrackingActions.PRODUCT_RESTRICTIONS_SHOP_SIMILAR}
					accessibilityLabel="Shop Similar Products Button"
				/>
			);
		}
	};

	renderBody = (restriction) => {
		const locales = restriction.locales.map((locale) => locale.stateCode);
		return (
			<View>
				<Text style={componentStyles.bodyText}>This item cannot be shipped to the following {pluralize('location', locales.length)}: {locales.join(', ')}</Text>
				{this.renderButton(restriction)}
			</View>
		);
	};

	renderRestrictions = () => {
		if (this.props.restrictions.length) {
			return this.props.restrictions.map((restriction, index) => {
				if (!Array.isArray(restriction.locales)) {
					return <View/>;
				}
				return (
					<InlineAlert
						key={index}
						title={restriction.policyDescription}
						style={componentStyles.inlineAlert}
					>
						{this.renderBody(restriction)}
					</InlineAlert>
				);
			});
		}
	};

	render() {
		return (
			<View>{this.renderRestrictions()}</View>
		);
	}

}

ProductRestrictions.propTypes = {
	application: PropTypes.string.isRequired,
	restrictions: PropTypes.array.isRequired,
	navigator: PropTypes.shape({
		push: PropTypes.func,
	}),
};

ProductRestrictions.defaultProps = {
	application: '',
	restrictions: [],
};

const mapStateToProps = (state, ownProps) => {
	const productComposite = productConfigurationHelpers.getProductComposite(ownProps.productConfigurationId) || {};
	const selectedFinish = productConfigurationHelpers.getSelectedFinish(ownProps.productConfigurationId) || {};
	return {
		application: productComposite.application,
		restrictions: selectedFinish.restrictions || [],
	};
};

export default connect(mapStateToProps)(ProductRestrictions);
