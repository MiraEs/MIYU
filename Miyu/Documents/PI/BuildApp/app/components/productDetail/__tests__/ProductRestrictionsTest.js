
jest.mock('../../InlineAlert', () => 'InlineAlert');
jest.mock('../../button', () => 'Button');
jest.mock('../../../lib/styles');
jest.mock('../../../lib/analytics/TrackingActions', () => ({}));
jest.mock('../../../lib/ProductConfigurationHelpers');
jest.mock('BuildNative');
jest.mock('pluralize');
jest.mock('react-redux');
jest.mock('../../../services/httpClient', () => ({}));

jest.unmock('react-native');

import React from 'react';
import 'react-native';
import { ProductRestrictions } from '../ProductRestrictions';
import renderer from 'react-test-renderer';

const fullProps = {
	application: 'application',
	restrictions: [{
		policyDescription: 'ab1953',
		locales: [{
			stateCode: 'CA',
		}],
	}],
	navigator: {
		push: jest.fn(),
	},
};

describe('ProductRestrictions', () => {

	it('should render with full props', () => {
		const wrapper = renderer.create(
			<ProductRestrictions {...fullProps} />
		);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});

	it('should handle navigateToSearch', () => {
		const wrapper = renderer.create(
			<ProductRestrictions {...fullProps} />
		);
		wrapper.getInstance().navigateToSearch();
		expect(fullProps.navigator.push).toBeCalledWith('productDrops', {
			searchCriteria: {
				keyword: 'application',
				page: 1,
				pageSize: 50,
			},
			tracking: {
				name: 'build:app:productrestrictions:searchresults',
				data: {
					keyword: 'application',
				},
			},
		});
	});

});
