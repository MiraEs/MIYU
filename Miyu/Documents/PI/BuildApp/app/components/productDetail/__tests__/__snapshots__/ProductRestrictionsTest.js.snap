// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRestrictions should render with full props 1`] = `
<View>
  <InlineAlert
    style={
      Object {
        "margin": 8,
      }
    }
    title="ab1953"
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#A94442",
            "textAlign": "center",
          }
        }
      >
        This item cannot be shipped to the following 
        : 
        CA
      </Text>
      <Button
        accessibilityLabel="Shop Similar Products Button"
        onPress={[Function]}
        style={
          Object {
            "marginTop": 8,
          }
        }
        text="Shop Similar Products"
        trackAction={undefined}
      />
    </View>
  </InlineAlert>
</View>
`;
